runtime/src/chpl-gpu.c

```diff
@@ -712,6 +718,7 @@ void chpl_gpu_arg_host_register(void* _cfg, void* arg, size_t size) {
   void *dev_arg = chpl_gpu_impl_host_register(*((void**)arg), size);
   *(cfg->host_registered_vars[cfg->cur_host_registered_var]) = dev_arg;
   cfg_add_direct_param(cfg, cfg->host_registered_vars[cfg->cur_host_registered_var]);
+  cfg->host_registered_vars_host_ptrs[cfg->cur_host_registered_var] = *((void**)arg);
```

Comment:
gah... so unfortunate that these APIs don't have an interface for getting a host pointer from a device one for this scenario...